--- smart-0.51/smart/channel.py~	2005-12-14 20:27:17.000000000 +0100
+++ smart-0.51/smart/channel.py	2007-07-30 19:13:30.203002200 +0200
@@ -296,8 +296,8 @@
     filenames = os.listdir(os.path.dirname(channels.__file__))
     infos = {}
     for filename in filenames:
-        if filename.endswith("_info.py"):
-            type = filename[:-8].replace("_", "-")
+        if filename.endswith("_info.pyc"):
+            type = filename[:-9].replace("_", "-")
             infos[type] = getChannelInfo(type)
     return infos

