--- smart-0.41/smart/backends/rpm/metadata.py~	2005-12-18 04:31:18.000000000 +0200
+++ smart-0.41/smart/backends/rpm/metadata.py	2006-03-04 15:02:39.000000000 +0200
@@ -22,7 +22,7 @@
 #
 from smart.cache import PackageInfo, Loader
 from smart.backends.rpm.base import *
-from smart.util import cElementTree
+import cElementTree
 from smart import *
 import posixpath
 import locale
--- smart-0.41/smart/channels/rpm_md.py~	2005-06-10 00:27:12.000000000 +0300
+++ smart-0.41/smart/channels/rpm_md.py	2006-03-04 15:28:46.000000000 +0200
@@ -21,7 +21,7 @@
 #
 from smart.backends.rpm.metadata import RPMMetaDataLoader
 from smart.util.filetools import getFileDigest
-from smart.util.elementtree import ElementTree
+from elementtree import ElementTree
 from smart.const import SUCCEEDED, FAILED, NEVER, ALWAYS
 from smart.channel import PackageChannel
 from smart import *
--- smart-0.41/setup.py~	2006-03-04 15:32:52.000000000 +0200
+++ smart-0.41/setup.py	2006-03-04 15:32:54.000000000 +0200
@@ -44,23 +44,6 @@
 config_h = sysconfig.get_config_h_filename()
 config_h_vars = sysconfig.parse_config_h(open(config_h))
 
-# cElementTree needed defines
-CET_DEFINES = [
-    ("XML_STATIC", None),
-    ("XML_NS", "1"),
-    ("XML_DTD", "1"),
-    ("XML_CONTEXT_BYTES", "1024")
-    ]
-if "HAVE_MEMMOVE" in config_h_vars:
-    CET_DEFINES.append(("HAVE_MEMMOVE", "1"))
-if "HAVE_BCOPY" in config_h_vars:
-    CET_DEFINES.append(("HAVE_BCOPY", "1"))
-if sys.byteorder == "little":
-    CET_DEFINES.append(("BYTEORDER", "1234"))
-else:
-    CET_DEFINES.append(("BYTEORDER", "4321"))
-
-
 setup(name="smart",
       version = VERSION,
       description = "Smart Package Manager is a next generation package "
@@ -87,7 +70,6 @@
                   "smart.interfaces.images",
                   "smart.plugins",
                   "smart.util",
-                  "smart.util.elementtree",
                  ],
       scripts = ["smart.py"],
       ext_modules = [
@@ -100,13 +82,6 @@
                                ["smart/util/ctagfile.c"]),
                      Extension("smart.util.cdistance",
                                ["smart/util/cdistance.c"]),
-                     Extension("smart.util.cElementTree",
-                               ["smart/util/celementtree/cElementTree.c",
-                                "smart/util/celementtree/expat/xmlparse.c",
-                                "smart/util/celementtree/expat/xmlrole.c",
-                                "smart/util/celementtree/expat/xmltok.c"],
-                                include_dirs=["smart/util/celementtree/expat"],
-                                define_macros=CET_DEFINES),
                     ],
       data_files = I18NFILES +
                    [(PYTHONLIB+"/smart/interfaces/images", 
--- smart-0.41/smart/option.py~	2005-07-19 23:12:55.000000000 +0300
+++ smart-0.41/smart/option.py	2006-03-04 16:26:11.000000000 +0200
@@ -19,7 +19,7 @@
 # along with Smart Package Manager; if not, write to the Free Software
 # Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 #
-from smart.util import optparse
+import optparse
 from smart import Error, _
 import textwrap
 import sys, os
