diff -Naur smart-0.50rc1.orig/smart/backends/rpm/base.py smart-0.50rc1/smart/backends/rpm/base.py
--- smart-0.50rc1.orig/smart/backends/rpm/base.py	2007-01-09 12:17:08.000000000 +0100
+++ smart-0.50rc1/smart/backends/rpm/base.py	2007-01-09 12:19:22.000000000 +0100
@@ -37,8 +37,6 @@
 except ImportError:
     raise Error, _("'rpm' python module is not available")
 
-archscore = rpm.archscore
-
 __all__ = ["RPMPackage", "RPMProvides", "RPMNameProvides", "RPMPreRequires",
            "RPMRequires", "RPMUpgrades", "RPMConflicts", "RPMObsoletes",
            "rpm", "getTS"]
@@ -186,8 +184,6 @@
                 otherver, otherarch = splitarch(other.version)
                 if selfver != otherver:
                     rc = vercmp(self.version, other.version)
-                if rc == 0:
-                    rc = -cmp(archscore(selfarch), archscore(otherarch))
         return rc == -1
 
 class RPMProvides(Provides):         __slots__ = ()
diff -Naur smart-0.50rc1.orig/smart/backends/rpm/header.py smart-0.50rc1/smart/backends/rpm/header.py
--- smart-0.50rc1.orig/smart/backends/rpm/header.py	2007-01-09 12:17:08.000000000 +0100
+++ smart-0.50rc1/smart/backends/rpm/header.py	2007-01-09 12:17:57.000000000 +0100
@@ -203,8 +203,6 @@
             if h[1106]: # RPMTAG_SOURCEPACKAGE
                 continue
             arch = h[1022] # RPMTAG_ARCH
-            if rpm.archscore(arch) == 0:
-                continue
 
             name = h[1000] # RPMTAG_NAME
             epoch = h[1003] # RPMTAG_EPOCH
diff -Naur smart-0.50rc1.orig/smart/backends/rpm/metadata.py smart-0.50rc1/smart/backends/rpm/metadata.py
--- smart-0.50rc1.orig/smart/backends/rpm/metadata.py	2007-01-09 12:17:08.000000000 +0100
+++ smart-0.50rc1/smart/backends/rpm/metadata.py	2007-01-09 12:18:27.000000000 +0100
@@ -160,10 +160,7 @@
                         skip = None
 
                 elif tag == ARCH:
-                    if rpm.archscore(elem.text) == 0:
-                        skip = PACKAGE
-                    else:
-                        arch = elem.text
+                    arch = elem.text
 
                 elif tag == NAME:
                     name = elem.text
diff -Naur smart-0.50rc1.orig/smart/backends/rpm/redcarpet.py smart-0.50rc1/smart/backends/rpm/redcarpet.py
--- smart-0.50rc1.orig/smart/backends/rpm/redcarpet.py	2007-01-09 12:17:08.000000000 +0100
+++ smart-0.50rc1/smart/backends/rpm/redcarpet.py	2007-01-09 12:18:47.000000000 +0100
@@ -191,10 +191,7 @@
         self._release = data
 
     def handleArchEnd(self, name, attrs, data):
-        if rpm.archscore(data) == 0:
-            self._skip = self.PACKAGE
-        else:
-            self._arch = data
+        self._arch = data
 
     def handleSectionEnd(self, name, attrs, data):
         self._info["group"] = data
diff -Naur smart-0.50rc1.orig/smart/backends/rpm/synthesis.py smart-0.50rc1/smart/backends/rpm/synthesis.py
--- smart-0.50rc1.orig/smart/backends/rpm/synthesis.py	2007-01-09 12:17:08.000000000 +0100
+++ smart-0.50rc1/smart/backends/rpm/synthesis.py	2007-01-09 12:20:11.000000000 +0100
@@ -164,9 +164,6 @@
                 else:
                     version, arch = version[:dot], version[dot+1:]
                 versionarch = "@".join((version, arch))
-                
-                if rpm.archscore(arch) == 0:
-                    continue
 
                 name = "-".join(rpmnameparts[0:-2])
 
diff -Naur smart-0.50rc1.orig/smart/backends/rpm/yast2.py smart-0.50rc1/smart/backends/rpm/yast2.py
--- smart-0.50rc1.orig/smart/backends/rpm/yast2.py	2007-01-09 12:17:08.000000000 +0100
+++ smart-0.50rc1/smart/backends/rpm/yast2.py	2007-01-09 12:19:51.000000000 +0100
@@ -175,8 +175,6 @@
                 nameparts = entryname.split(" ")
                 # skip entry if arch is not compatible
                 arch = nameparts[3]
-                if rpm.archscore(arch) <= 0:
-                    return
                 name = nameparts[0]
                 self.curpkgname = name
                 version = nameparts[1]
